x-pack/plugins/maps/public/classes/sources/vector_source/mvt_vector_source.ts

```diff
@@ -25,4 +28,18 @@ export interface IMvtVectorSource extends IVectorSource {
    * Use getTileSourceLayer to specify the displayed source layer.
    */
   getTileSourceLayer(): string;
+
+  /**
+   * Syncs source specific styling with mbMap this allows custom sources to further style the map layers/filters
+   */
+  syncSourceStyle?(mbMap: MbMap, getLayerIds: () => string[]): void;
+
+  /**
+   * specifies if a source provides its own legend details or if the default vector_style is used if the source has this method it must also implement renderLegendDetails
+   */
+  hasLegendDetails?(): Promise<boolean>;
```

Comment:
I think hasLegendDetails and renderLegendDetails can state in IVectorSource. I was only suggesting moving syncSourceStyle to IMvtVectorSource since its only used in MvtVectorLayer.